adapters/yandex/yandex.go

```diff
@@ -0,0 +1,347 @@
+func (a *adapter) resolveUrl(placementID yandexPlacementID, referer string, currency string) (string, error) {
+	params := macros.EndpointTemplateParams{PageID: placementID.PageID}
+
+	endpointStr, err := macros.ResolveMacros(a.endpoint, params)
+	if err != nil {
+		return "", err
+	}
+
+	parsedUrl, err := url.Parse(endpointStr)
+	if err != nil {
+		return "", err
+	}
+
+	addNonEmptyQueryParams(parsedUrl, map[string]string{
+		refererQueryKey:  referer,
+		currencyQueryKey: currency,
+		impIdQueryKey:    placementID.ImpID,
+	})
+
+	return parsedUrl.String(), nil
+}
+
+func addNonEmptyQueryParams(url *url.URL, queryMap map[string]string) {
+	query := url.Query()
+	for key, value := range queryMap {
+		if len(value) > 0 {
+			query.Add(key, value)
+		}
+	}
+
+	url.RawQuery = query.Encode()
+}
+
+func getReferer(request *openrtb2.BidRequest) string {
+	if request.Site == nil {
+		return ""
+	}
+
+	return request.Site.Domain
+}
+
+func getCurrency(request *openrtb2.BidRequest) string {
+	if len(request.Cur) == 0 {
+		return ""
+	}
+
+	return request.Cur[0]
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, _ *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+
+	if adapters.IsResponseStatusCodeNoContent(responseData) {
+		return nil, nil
+	}
+
+	if err := adapters.CheckResponseStatusCodeForErrors(responseData); err != nil {
+		return nil, []error{err}
+	}
+
+	var bidResponse openrtb2.BidResponse
+	if err := json.Unmarshal(responseData.Body, &bidResponse); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Bad server response: %d", err),
+		}}
+	}
+
+	bidResponseWithCapacity := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+
+	var errors []error
+
+	impMap := map[string]*openrtb2.Imp{}
+	for i := range request.Imp {
+		imp := request.Imp[i]
+
+		impMap[imp.ID] = &imp
+	}
+
+	for _, seatBid := range bidResponse.SeatBid {
+		for i := range seatBid.Bid {
+			bid := seatBid.Bid[i]
+
+			imp, exists := impMap[bid.ImpID]
+			if !exists {
+				errors = append(errors, &errortypes.BadInput{
+					Message: fmt.Sprintf("Invalid bid imp ID #%s does not match any imp IDs from the original bid request", bid.ImpID),
+				})
+				continue
+			}
+
+			bidType, err := getBidType(*imp)
+			if err != nil {
+				errors = append(errors, err)
+				continue
+			}
+
+			bidResponseWithCapacity.Bids = append(bidResponseWithCapacity.Bids, &adapters.TypedBid{
+				Bid:     &bid,
+				BidType: bidType,
+			})
+		}
+	}
+
+	return bidResponseWithCapacity, errors
+}
+
+func getBidType(imp openrtb2.Imp) (openrtb_ext.BidType, error) {
+	if imp.Native != nil {
+		return openrtb_ext.BidTypeNative, nil
+	}
+
+	if imp.Banner != nil {
+		return openrtb_ext.BidTypeBanner, nil
+	}
+
+	if imp.Audio != nil {
+		return openrtb_ext.BidTypeAudio, nil
+	}
+
+	if imp.Video != nil {
+		return openrtb_ext.BidTypeVideo, nil
+	}
```

Comment:
Can we remove these checks since the adapter only supports banner and native media types?

Reply:
Sure, will do!